packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -299,6 +306,9 @@ export class EventSourceMapping extends cdk.Resource implements IEventSourceMapp
       throw new Error(`tumblingWindow cannot be over 900 seconds, got ${props.tumblingWindow.toSeconds()}`);
     }
 
+    if (props.startingPosition === StartingPosition.AT_TIMESTAMP && !props.startingPositionTimestamp) {
+      throw new Error('startingPositionTimestamp must be provided when startingPosition is AT_TIMESTAMP');
+    }
```

Comment:
what about when `startingPositionTimestamp` is set but `props.startingPosition !== StartingPosition.AT_TIMESTAMP`? I think we should throw an error also

Reply:
Added another error and a test